awx/main/analytics/collectors.py

```diff
@@ -280,7 +288,20 @@ def _copy_table(table, query, path):
     return file.file_list()
 
 
-@register('events_table', '1.2', format='csv', description=_('Automation task records'), expensive=True)
+def events_slicing(key, since, until):
+    from awx.conf.models import Setting
+
+    step = 100000
+    last_entries = Setting.objects.filter(key='AUTOMATION_ANALYTICS_LAST_ENTRIES').first()
+    last_entries = last_entries.value if last_entries is not None else {}
+    previous_pk = last_entries.get(key) or 0
+    final_pk = models.JobEvent.objects.filter(created_lte=until).aggregate(Max('pk'))['pk__max']
```

Comment:
Should we still limit the `previous_pk` to max 4 weeks back like the others? If tower can take it, I don't mind sending the whole gap (e.g. if customer stops the collection for a year and then starts it when they'll see some new useful feature)

But if we'd be sending the whole gap, we'll need to send also the other collections. E.g. analytics will not use events that can't be associated to jobs (since we're grouping by job attrs like org, inventory, etc.). 

@ryanpetrello thoughts?

Reply:
(also if we're not setting last 4 weeks, we should at least set MIN(id), instead of 0, or this will run for a while from 0. If e.g. the customer produces couple of B events every week and deletes them. )

Reply:
Ok, I've updated this.